src/internal/m365/collection/drive/helper_test.go

```diff
@@ -996,6 +1022,78 @@ func (dd *deltaDrive) newEnumer() *DeltaDriveEnumerator {
 	return &DeltaDriveEnumerator{Drive: clone}
 }
 
+type drivePrevPaths struct {
+	id                 string
+	folderIDToPrevPath map[string]string
+}
+
+func (dd *deltaDrive) newPrevPaths(
+	idPathPairs ...string,
+) *drivePrevPaths {
+	dpp := drivePrevPaths{
+		id:                 dd.id,
+		folderIDToPrevPath: map[string]string{},
+	}
+
+	if len(idPathPairs)%2 == 1 {
+		dpp.folderIDToPrevPath["error"] = "idPathPairs had odd count of elements"
+		return &dpp
+	}
+
+	for i := 0; i < len(idPathPairs); i += 2 {
+		dpp.folderIDToPrevPath[idPathPairs[i]] = idPathPairs[i+1]
+	}
+
+	return &dpp
+}
+
+// transforms 0 or more drivePrevPaths to a map[driveID]map[folderID]prevPathString
+func multiDrivePrevPaths(drivePrevs ...*drivePrevPaths) map[string]map[string]string {
+	msmss := map[string]map[string]string{}
```

Comment:
nit: could you please use a more descriptive name? Or at least one where the reader isn't tempted to try to figure out what the random letters mean (I'm guess mapStringMapStringString?)? In my opinion even `res` would be more useful here because at least that has meaning. `msmss` doesn't have clear meaning just from looking at it

Reply:
Can I name it `missMisses`?

This one gets updated in the next PR.